membershiprelation.php

```diff
@@ -52,6 +52,37 @@ function membershiprelation_civicrm_post($op, $objectName, $objectId, &$objectRe
     }
   }
 
+  function membershiprelation_civicrm_pre($op, $objectName, $objectId, &$params) {
+    if ($op == "edit" && $objectName == "Membership") {
+      if (!empty($params['membership_type_id'])) {
+        if (in_array($params['membership_type_id'], [VMTONEGIRL, VMTTWOGIRLS, VMTTHREEGIRLS, VMTFOURGIRLS])) {
+          $membershipObj = new CRM_Member_DAO_Membership();
+          $membershipObj->id = $objectId;
+          $membershipObj->find();
+          $oldType = NULL;
+          while ($membershipObj->fetch()) {
+            $oldType = $membershipObj->membership_type_id;
+          }
+          if (!in_array($oldType, [VMTONEGIRL, VMTTWOGIRLS, VMTTHREEGIRLS, VMTFOURGIRLS])) {
+            unset($params['id']);
+          }
+        }
+        else {
+          $membershipObj = new CRM_Member_DAO_Membership();
```

Comment:
This section doesn't make as much sense to me primarily due to my comment below

Reply:
Else part is for a chapter or any membership type other then virtual membership. And below at L78 we are ensuring that the old membership is not any of the virtual memberrship, if the old membership in VM, then unset the id so that we can keep the virtual membership instact w/o updating its start or end date.